src/analysis/normed_space/units.lean

```diff
@@ -288,3 +288,42 @@ lemma open_embedding_coe : open_embedding (coe : Rˣ → R) :=
 open_embedding_of_continuous_injective_open continuous_coe ext is_open_map_coe
 
 end units
+
+section ideal
+
+/-- An ideal which contains an element within `1` of `1 : R` is `⊤`-/
+lemma ideal.eq_top_of_norm_lt_one (I : ideal R) {x : R} (hxI : x ∈ I) (hx : ∥1 - x∥ < 1) : I = ⊤  :=
+let u := units.one_sub (1 - x) hx in (I.eq_top_iff_one.mpr $
+  by simpa only [show u.inv * x = 1, by simp] using I.mul_mem_left u.inv hxI)
+
+variable (R)
+
+/-- The `nonunits` in a complete normed ring are contained in the complement of the ball of radius
+`1` centered at `1 : R`. -/
+lemma nonunits.subset_compl_ball : nonunits R ⊆ (metric.ball (1 : R) 1)ᶜ :=
+set.subset_compl_comm.mp $ λ x hx, by simpa [sub_sub_self, units.coe_one_sub] using
+  (units.one_sub (1 - x) (by rwa [metric.mem_ball, dist_eq_norm, norm_sub_rev] at hx)).is_unit
+
+variable {R}
+
+/-- A proper `ideal` in a complete normed ring is contained in the complement of the ball of radius
+`1` centered at `1 : R`. -/
+lemma ideal.subset_compl_ball_of_ne_top (I : ideal R) (hI : I ≠ ⊤) :
+  (I : set R) ⊆ (metric.ball (1 : R) 1)ᶜ :=
+(coe_subset_nonunits hI).trans $ nonunits.subset_compl_ball R
+
+/-- The `ideal.closure` of a proper `ideal` in a complete normed ring is proper. -/
```

Comment:
```suggestion
/-- The `ideal.closure` of a proper ideal in a complete normed ring is proper. -/
```
since you don't put ` ` around "ideal" at most places.